pkg/cloudprovider/providers/aws/device_allocator.go

```diff
@@ -79,6 +100,19 @@ func (d *deviceAllocator) GetNext(existingDevices ExistingDevices) (mountDevice,
 	}
 }
 
+func (d *deviceAllocator) LockForUse() {
+	d.deviceLock.Lock()
+}
+
+func (d *deviceAllocator) MarkForUse() {
```

Comment:
I think the caller knows the device, so can we have the function take the device, and `panic` if it doesn't match what we expect?   Even if it's correct today, it's liable to get broken in future.

It's also a more self-explanatory API